### ISSUE TITLE
[BUG] Typescript bindings fieldOrder

### ISSUE THREAD

**Issue by starknetdev** [dojoengine/dojo#2748]
**Describe the bug**
`fieldOrder` is added to all models and struct types stored within the models in `models.gen.ts` which I assume is necessary for querying torii. The current problem is that `contracts.gen.ts` references these for inputs, but `fieldOrder` should not be present otherwise there are deserialization errors.

**To Reproduce**

* Clone [tournament_component](https://github.com/Provable-Games/arcade-components/tree/main)
* run, cd tournaments && cd contracts && sozo build --typescript
* inspect, contracts/bindings/typescript/models.gen.ts + contracts.gen.ts

or find the file here:

* [models.gen.ts](https://github.com/Provable-Games/arcade-components/tree/main/tournaments/contracts/bindings/typescript/models.gen.ts)
* [contracts.gen.ts](https://github.com/Provable-Games/arcade-components/tree/main/tournaments/contracts/bindings/typescript/models.gen.ts)

**Current Behaviour**
`models.gen.ts`
```
export interface Adventurer {
  fieldOrder: string[];
  ...fields
}
```
`contracts.gen.ts`
```
  const loot_survivor_mock_setAdventurer = async (
    adventurerId: BigNumberish,
    adventurer: models.Adventurer
  ) => {
    try {
      return await provider.call("tournament", {
        contractName: "loot_survivor_mock",
        entrypoint: "set_adventurer",
        calldata: [adventurerId, adventurer],
      });
    } catch (error) {
      console.error(error);
    }
  };
  ```

**Expected behavior**

The binding can add a type that removes `fieldOrder` (and from nested objects), or handle this is in the generation process. 

`models.gen.ts`
```
export interface Adventurer {
  fieldOrder: string[];
  ...fields
}

// Utility type to recursively remove fieldOrder from all nested objects
type RemoveFieldOrder<T> = T extends object
  ? Omit<
      {
        [K in keyof T]: T[K] extends object ? RemoveFieldOrder<T[K]> : T[K];
      },
      'fieldOrder'
    >
  : T;

// Create a new type without fieldOrder
type InputAdventurer = RemoveFieldOrder<Adventurer>;
```
`contracts.gen.ts`
```
  const loot_survivor_mock_setAdventurer = async (
    adventurerId: BigNumberish,
    adventurer: models.InputAdventurer
  ) => {
    try {
      return await provider.call("tournament", {
        contractName: "loot_survivor_mock",
        entrypoint: "set_adventurer",
        calldata: [adventurerId, adventurer],
      });
    } catch (error) {
      console.error(error);
    }
  };
  ```

**Comment by MartianGreed** [comment]
Thank's for this valuable feedbac, will add this to current PR

**Comment by rsodre** [comment]
or maybe `fieldOrder` can be removed from all models, and use a generic type when needed, like this:

```ts
export interface Score {
  points: number;
};

export type ModelWithFieldOrder<T> = T & {
  fieldOrder: string[];
};

export const _score: Score = {
  points: 0,
};
export const _score_with_field_order: ModelWithFieldOrder<Score> = {
  fieldOrder: ['honour'],
  ..._score,
};
```

**Comment by glihm** [comment]
Merged in #2773.
